eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -125,13 +184,46 @@ stages:
               Commitish: $(SpecRepoCommit)
               WorkingDirectory: $(SpecRepoDirectory)
             - Name: $(SdkRepoOwner)/$(SdkRepoName)
-              Commitish: ${{ parameters.SdkRepoCommit }}
+              Commitish: $(CheckoutCommitish)
               WorkingDirectory: $(SdkRepoDirectory)
           SkipCheckoutNone: true
           ${{ if and(eq(variables['System.TeamProject'], 'internal'), endsWith(variables['Build.Repository.Name'], '-pr')) }}:
             TokenToUseForAuth: $(azuresdk-github-pat)
             PreserveAuthToken: true
 
+      - ${{ if and(eq(variables['System.TeamProject'], 'internal'), eq(parameters.CreatePullRequest, true), ne(variables['Build.Reason'], 'PullRequest')) }}:
+        - pwsh: |
```

Comment:
I don't think we need this step because https://github.com/Azure/azure-rest-api-specs/blob/main/eng/common/scripts/git-branch-push.ps1#L70 already handles this work.

Reply:
Updated.